elastic/fleet-server
# Add diagnostics config hook

## What is the problem this PR solves?

fleet-server.yml is not part of the agent diagnostics bundle

## How does this PR solve the problem?

Add a diagnostics hook to the agent client that dumps the current  (redacted) config.

## How to test this PR locally

Run an agent with a fleet-server binary built from this commit.
Request diagnostics through the UI/terminal
Unarchive the bundle and ensure that `components/fleet-server-default/fleet-server.yml` exists and is not empty

## Design Checklist

- ~~I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.~~
- ~~I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.~~
- ~~I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.~~

## Checklist



- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~
- ~~I have added tests that prove my fix is effective or that my feature works~~

## Related issues

- Closes #2623
